pkg/store/config/config.go

```diff
@@ -0,0 +1,24 @@
+package config
+
+import (
+	"errors"
+	"strings"
+)
+
+type StoreConfig struct {
+	URL           string `split_words:"true" required:"true"`
+	ReindexOnBoot bool   `split_words:"true" default:"false"`
+	Insecure      bool   `split_words:"true" default:"false"`
+	CertPath      string `split_words:"true"`
+	PoolPath      string `split_words:"true"`
+}
+
+func (c StoreConfig) Validate() error {
+	// If the insecure flag isn't set then we must have certs when connecting to trtl.
+	if strings.HasPrefix(c.URL, "trtl://") && !c.Insecure {
+		if c.CertPath == "" || c.PoolPath == "" {
+			return errors.New("invalid configuration: connecting to trtl over mTLS requires certs and cert pool")
+		}
+	}
+	return nil
+}
```

Comment:
Moved this configuration to a separate package to avoid the import cycle, so now this can be imported by both GDS and the BFF.

Reply:
Makes sense - I'll have to remember that this will change the configuration during the deployment.